server.js

```
@@ -0,0 +1,221 @@
+                        };
+                        let htmlEnd = `    </ol>
+                    </body>
+                    
+                    </html>`;
+                        fs.writeFile('./public/index.html', newIndex + htmlEnd, (err) => {
+                            if (err) {
+                                throw err;
+                            }
+                            console.log('Index.html updated!');
+                        });
+                    });
+                });
+                res.end();
+            });
+        };
+        // PUT METHOD
+        if (req.method === 'PUT') {
+            fs.readdir(`./public/`, (err, files) => {
+                if (files.includes(URL.slice(1)) === false) {
+                    res.writeHead(500, {
+                        'Date': `${date}`,
+                        'Content-Type': 'application/json',
+                    });
+                    res.write(`{ "error" : "resource ${URL} does not exist" }`);
+                    res.end()
+                } else {
+                    let parsedInfo = querystring.parse(body);
+                    const template = `<!DOCTYPE html>
+                <html lang="en">
+                <head>
+                <meta charset="UTF-8">
+                <title>The Elements - ${parsedInfo.elementName}</title>
+                <link rel="stylesheet" href="/styles.css">
+                </head>
+                <body>
+                    <h1>${parsedInfo.elementName}</h1>
+                    <h2>${parsedInfo.elementSymbol}</h2>
+                    <h3>${parsedInfo.elementAtomicNumber}</h3>
+                    <p>${parsedInfo.elementDescription}</p>
+                    <p><a href="/">back</a></p>
+                </body>
+                </html>`;
+                    fs.writeFile(`./public/${parsedInfo.elementName}.html`, template, (err) => {
+                        if (err) {
+                            return console.log(err);
```

This should return an error to the client using either a 400/500 error